scripts/uosc/elements/Menu.lua

```diff
@@ -813,13 +813,26 @@ function Menu:search_query_update(query, menu)
 end
 
 ---@param event? string
-function Menu:search_backspace(event)
+---@param word_mode? boolean Delete by words.
+function Menu:search_backspace(event, word_mode)
 	local pos, old_query, is_palette = #self.current.search.query, self.current.search.query, self.current.palette
-	-- The while loop is for skipping utf8 continuation bytes
-	while pos > 1 and old_query:byte(pos) >= 0x80 and old_query:byte(pos) <= 0xbf do
+	if word_mode then
+		local word_pat, other_pat = '[^%c%s%p]+$', '[%c%s%p]+$'
+		local init_pat = old_query:sub(#old_query):match(word_pat) and word_pat or other_pat
+		-- First we match all same type consecutive chars at the end
+		local tail = old_query:match(init_pat)
+		-- If there's only one, we extend the tail with opposite type chars
+		if tail and #tail == 1 then
+			tail = tail .. old_query:sub(1, #old_query - #tail):match(init_pat == word_pat and other_pat or word_pat)
+		end
+		pos = pos - #tail
```

Comment:
I would have expected `ctrl+backspace` to always delete back to the beginning of the last word. From a quick test at least KDE things seem to consistently behave that way.
```suggestion
		pos = old_query:find('[^%c%s%p]*[%c%s%p]*$') - 1
```

Reply:
Apparently that's not the case for vscode, it behaves like your original code. It doesn't really matter i guess.

Reply:
Yeah I've tried making it like vscode, which is how I expected it to work from the beginning. 

For example, if I have a string like:

```
abc.--
```

I'd want `ctrl+backspace` to end before `c`, not delete the whole string.

vscode has a bit more conditional behaviors, but I didn't want to make it that complex.

Reply:
> I would have expected `ctrl+backspace` to always delete back to the beginning of the last word.

This behavior is iconsistent across applications, unfortunately. Sublime Text, for example, deletes until the next boundary between word characters, punctuation, or whitespace.